halo2_proofs/src/dev.rs

```diff
@@ -513,6 +520,10 @@ impl<F: FieldExt> MockProver<F> {
         let permutation = permutation::keygen::Assembly::new(n, &cs.permutation);
         let constants = cs.constants.clone();
 
+        let challenges = iter::repeat_with(|| F::random(OsRng))
+            .take(cs.num_challenges)
+            .collect();
```

Comment:
This helps ensure the rest of `MockProver` continues to work, but it adds non-determinism to `MockProver` (that isn't present in the real prover). I'd like to avoid that if possible...

Reply:
Yeah, the hash chain looks fine to me. What precisely we do here is an implementation detail that can be altered at any time.